### ISSUE TITLE
Deleting Groups with active KeeShare sync configuration and recreate of them, deletes all entries in the original db

### ISSUE THREAD

**Issue by mgeramb** [keepassxreboot/keepassxc#7300]
**Steps to reproduce:**

- Create new KeyPass DB with name "Shared"
- Save it as "Shared.kdbx"
- Create one Entry "Test" in the "Shared" DB
- Close DB
- Create new KeyPass DB with name "MyDb"
- Save it as "MyDb.kdbx"
- Create new Group 'Sync Shared' in "MyDB" and select KeeShare
- Select "Syncronize" as type, enter the path of "Shared.kdbx" and enter the password
- Close the dialog with "Ok" button. The "Test" entry will be shown in the group "Sync Shared"
- Delete the group "Sync Shared". The group will now be shown in the Recycle Bin
- Delete the group in the Recylce Bin
- Create new Group 'Sync Shared' in "MyDB" and select KeeShare
- Select "Syncronize" as type, enter the path of "Shared.kdbx" and enter the password
- Close the dialog with "Ok" button

**Current Result:**
The "Test" entry will not be shown in the Group "Sync Shared"
If you open the DB "Shared.kdbx" the "Test" entry is also missing there.

**Expected Result:**
The "Test" entry should be shown in the group "Sync Shared" and in the "Shared" database

**Comment by droidmonkey** [comment]
This is the exact issue described in #6477

**Comment by mgeramb** [comment]
@droidmonkey : 
I'am not sure if this is exactly the same issue descripted in https://github.com/keepassxreboot/keepassxc/issues/6477, Maybe the technical reason in the background is the same, but I not read anything about deleting the whole shared group. This was the reason for me to provide this information. If you know that this has the same technical background, you can close this issue.

My comment in https://github.com/keepassxreboot/keepassxc/issues/6477 is another one. This is only related to the current 2.7 beta version and it seems that it has an other technical reason.

**Comment by droidmonkey** [comment]
Yes it's the same technical reason, but your description and recreate steps are really good. The reason any of this happens is because when you delete an object from your database, a deleted object uuid is stored forever in the kdbx backend for merging purposes. The problem arises with keyshare because you reintroduce an entry with the same uuid as the one you deleted, the merge algorithm says it needs to be deleted. 

The fix I have in the pipeline is to give entries new uuids if they are moved outside the share group. Basically treat the share group as an internal database. Right now the databases essentially get flattened into each other which causes these problems.

**Comment by speendo** [comment]
For the time this issue is not solved in KeePassXC I wrote a python program that is designed to remove all <DeletedObject> entries from the main database and all databases connected via KeeShare.

Please use it (at own risk) and tell me if it was useful and/or if anything can be improved.

Find it here: https://github.com/speendo/RemoveKeePass2DeletedObjects
